File: Billing.Api/Models/Salesman.cs
using System;

namespace Billing.Api.Models;

public class Salesman
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;
    public DateTime DateOfBirth { get; set; }
    public string Address { get; set; } = "";
    public string City { get; set; } = "";
    public string Mobile { get; set; } = "";

    public DateTime DateOfJoin { get; set; }

    public bool IsActive { get; set; } = true;
}

File: billing-ui/src/models/Salesman.ts
export interface SalesmanDto {
    id: number;
    name: string;
    dob?: string;
    address?: string;
    city?: string;
    mobile: string;
    dateOfJoin?: string;
    isActive: boolean;
}

export interface SalesmanCompensationSummaryDto {
    salesmanId: number;
    salesmanName: string;
    totalSalary: number;
    totalAdvance: number;
    netPaid: number;
}

export interface SalesmanCompensationDetailDto {
    id: number;
    entryDate: string;
    entryType: string;
    amount: number;
    source: string;
}
